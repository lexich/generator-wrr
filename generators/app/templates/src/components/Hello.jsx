"use strict";

import React, { PropTypes } from "react";
import styleCSS from "./Hello.css";

export default class Hello extends React.Component {
  static propTypes = {
    style: PropTypes.object.isRequired
  };
  static defaultProps = {
    style: styleCSS
  };
  render() {
    const { children, style } = this.props;
    return (
      <span className={style.root}>
        Hello
      </span>
    );
  }
}
