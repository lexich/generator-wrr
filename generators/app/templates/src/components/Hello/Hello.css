.root {
  color: blue;
}

.image {
  min-width: 100px;
  max-width: 400px;
  width: 100%;
}
