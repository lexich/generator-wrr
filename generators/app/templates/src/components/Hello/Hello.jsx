"use strict";

import React, { PropTypes } from "react";
import styleCSS from "./Hello.css";
import imageSeaLion from "./SeaLion.svg";


export default class Hello extends React.Component {
  static propTypes = {
    className: PropTypes.string,
    style: PropTypes.object.isRequired,
    src: PropTypes.string.isRequired
  };
  static defaultProps = {
    style: styleCSS,
    src: imageSeaLion
  };
  render() {
    const { style, src } = this.props;

    const className = style.root +
      (this.props.className ? ` ${this.props.className}` : "");

    return (
      <span className={className}>
        <img className={style.image} src={src} role="presentation" />
      </span>
    );
  }
}
