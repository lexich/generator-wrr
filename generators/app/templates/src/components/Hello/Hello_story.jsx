import React from "react";
import { storiesOf } from "@kadira/storybook";
import Hello from ".";

storiesOf("Hello", module)
  .add("Hello", () => (
    <Hello />
  ))
  .add("Hello with src", ()=> (
    <Hello src={"http://placehold.it/350x150"} />
  ));
