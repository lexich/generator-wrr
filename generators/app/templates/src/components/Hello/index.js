"use strict";

import Hello from "./Hello";

export default Hello;
