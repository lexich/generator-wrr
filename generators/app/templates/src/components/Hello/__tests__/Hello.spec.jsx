/* globals describe, it, expect */
import renderer from "react-test-renderer";
import React from "react";
import Hello from "../";

describe("Hello component", ()=> {
  it("default args", ()=> {
    const component = renderer.create(
      <Hello />
    );
    expect(component).toMatchSnapshot();
  });

  it("check src args", ()=> {
    const component = renderer.create(
      <Hello src="best-seal.jpg" />
    );
    expect(component).toMatchSnapshot();
  });
});
