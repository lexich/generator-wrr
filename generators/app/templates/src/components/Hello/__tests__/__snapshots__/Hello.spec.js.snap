exports[`Hello component check src args 1`] = `
<span
  className="root">
  <img
    className="image"
    role="presentation"
    src="best-seal.jpg" />
</span>
`;

exports[`Hello component default args 1`] = `
<span
  className="root">
  <img
    className="image"
    role="presentation"
    src="filepath.mock" />
</span>
`;
